### DISCUSSION TITLE
Is there a way to increase the saveTimeout value in DocumentsMainImpl?

### DISCUSSION THREAD

**Discussion by safisa** [eclipse-theia/theia#14550] [Q&A] [answered]
Hi,
I am using vscode.workspace.onWillSaveTextDocument in the resolveCustomTextEditor (vscode.CustomTextEditorProvider), this method has a saveTimeout of 1750ms in the saveTimeout class in Theia.
Sometimes the save operation takes more than this timeout value and it is rejected in the DocumentsMainImpl class.
Is there a way to override this timeout?

Thanks

**Comment by msujew** [comment, marked as answer]
Hey @safisa,

I believe the timeout stems from VS Code itself, see [here](https://github.com/microsoft/vscode/blob/38d96d1dd5562c5b4c027f15f7262e83db5a721c/src/vs/workbench/api/browser/mainThreadSaveParticipant.ts#L35-L38). It is very hardcoded in there.

Assuming you get somehow access to the `RPCProtocol` object in the browser frontend, you should be able to get the `DocumentsMainImpl` instance and adjust the timeout, given that it's just a protected field.

However, I don't think there's an intended way of adjusting the timeout. Feel free to contribute a PR to improve this :)
